crt_portal/cts_forms/forms.py

```diff
@@ -1322,6 +1323,16 @@ class BulkActions(Form, ActivityStreamUpdater):
         ),
     )
 
+    def __init__(self, query):
+        Form.__init__(self)
+        keys = ['assigned_section', 'status', 'primary_statute', 'district']
+        values = query.values_list(*keys)
+        initial_values = list(zip(*values))
+        for index, key in enumerate(keys):
+            initial = initial_values[index]
+            if len(set(initial)) == 1:
+                self.fields[key].initial = initial[0]
```

Comment:
@Jkrzy I'm not crazy about this block of code, it's fairly complicated and it feels like we can do some fancy database magic to get these results quickly. Maybe using annotations? Open to ideas.

Reply:
This is a great candidate to be moved into it's own function with a docstring to both clarify the intent, facilitate testing and evolution to something you're more comfortable with.

Nothing immediately comes to mind from the Django ORM that would get us exactly what we're looking for -- Are all values for `X` the same across our QuerySet? -- in a single query

Breaking it into one query for each field is an option, straightforward, and avoids potential problems with the inbound queryset being huge. Something like

```python
values = Report.objects.all().values_list('status', flat=True).distinct()

print(values.query)
# SELECT DISTINCT "cts_forms_report"."status" FROM "cts_forms_report"

initial = values[0] if len(values) == 1 else None

```

Reply:
Updated with a simpler implementation, thanks. `distinct` doesn't seem to work here because the queryset may already have filters applied to it previously. 

For instance:

```sql
SELECT DISTINCT "cts_forms_report"."district", "cts_forms_report"."create_date" FROM "cts_forms_report" WHERE "cts_forms_report"."status" IN (new, open) ORDER BY "cts_forms_report"."create_date" DESC
```

Gives me `[None, 2, 2, '', None, None ...]`

I think we'd have to force the evaluation of the queryset, I tried doing `all()` but that just returns a copy. I may be missing something here, though.

Reply:
Ah, I think the `order_by` is the culprit -- https://docs.djangoproject.com/en/2.2/ref/models/querysets/#django.db.models.query.QuerySet.distinct

I don't _think_ we care about order here so this may work:
```python
query.order_by()...
```

Reply:
You rock! That was exactly it.